docs/en/advanced_guides/customize_transforms.md

```diff
@@ -1,3 +1,297 @@
 # Customize Data Transformation and Augmentation
 
-Coming soon.
+## DATA TRANSFORM
+
+In the OpenMMLab repositories, dataset construction and data preparation are decoupled from each other. Usually, the dataset construction only parses the dataset and records the basic information of each sample, while the data preparation is performed by a series of data transforms, such as data loading, preprocessing, and formatting based on the basic information of the samples.
+
+### To use Data Transforms
+
+In MMEngine, we use various callable data transforms classes to perform data manipulation. These data transformation classes can accept several configuration parameters for instantiation and then process the input data dictionary by calling. Also, all data transforms accept a dictionary as input and output the processed data as a dictionary. A simple example is as belows:
+
+> In MMEngine, we don’t have the implementations of data transforms. you can find the base data transform class and many other data transforms in MMCV. So you need to install MMCV before learning this tutorial, see the [MMCV installation guide](https://mmcv.readthedocs.io/en/2.x/get_started/installation.html).
+
+```python
+import numpy as np
+from mmcv.transforms import Resize
+transform = Resize(scale=(224, 224))
+data_dict = {'img': np.random.rand(256, 256, 3)}
+data_dict = transform(data_dict)
+print(data_dict['img'].shape)
+```
+
+```shell
+(224, 224, 3)
+```
+
+### Implement a simple custom data transformation
+
+To achieve custom data transformation, the following steps are usually required:
+
+1. Define a data augmentation class and register it in the `TRANSFORMS` register of `mmcv`
+
+2. Let the enhancement class inherit from `BaseTransform`
+
+3. Implement the `transform` function and have it accept input of dictionary type and return output of dictionary type
+   Taking the implementation of data augmentation as an example:
+
+```Python
+import random
+import mmcv
+from mmcv.transforms import BaseTransform, TRANSFORMS
+
+@TRANSFORMS.register_ Module() # 1 register
+class MyFlip (BaseTransform): # 2 Inherit BaseTransform
+    def __init__ (self, direction: str):
+        super().__init__ ()
+        Self. direction=direction
+    def transform (self, results: dict) ->dict: # Accept dictionary type input
+        img=results ['img ']
+        results ['img ']=mmcv. imflip (img, direction=self. direction) # Accept dictionary type output
+    return results
```

Comment:
Thank you for your contribution! Could you please refine the code format in these examples?